cabal-install/src/Distribution/Client/ProjectPlanning/Types.hs

```diff
@@ -750,18 +782,39 @@ data BuildStyle =
     -- Typically 'BuildAndInstall' packages will only depend on other
     -- 'BuildAndInstall' style packages and not on 'BuildInplaceOnly' ones.
     --
-  | BuildInplaceOnly
+  | BuildInplaceOnly MemoryOrDisk
+    -- | Built in-memory only using GHC multi-repl, they are not built or installed
+    -- anywhere on disk. BuildInMemory packages can't be depended on by BuildAndInstall nor BuildInplaceOnly packages
+    -- (because they don't exist on disk) but can depend on other BuildStyles.
+
+    -- At the moment BuildInMemory is only used by the 'repl' command.
+--  | BuildInMemory
   deriving (Eq, Show, Generic)
 
-instance Binary BuildStyle
-instance Structured BuildStyle
+data MemoryOrDisk = InMemory | OnDisk  deriving (Eq, Show, Generic)
```

Comment:
```suggestion
-- | Built in-memory only using GHC multi-repl, they are not built or installed
-- anywhere on disk. MemoryOrDisk packages can't be depended on by BuildAndInstall nor BuildInplaceOnly packages
-- (because they don't exist on disk) but can depend on other BuildStyles.
--
-- At the moment 'InMemory' is only used by the 'repl' command.
data MemoryOrDisk = InMemory | OnDisk  deriving (Eq, Show, Generic)
```

Reply:
Not sure what this is suggesting, this comment already exists to document `BuildInplaceOnly`.

Reply:
I thought the comment didn't move correctly. Discard if it was at the right location.